feat(profiling): add browser profiling docs

JonasBa opened the pull request:
Will require the latest release of browser SDK hence why the @TODO for version number. @indragiek lmk if the docs are too conservative, I tried leaving the disclaimer which I think is fair and something folks should be aware of. 

We are also not adding other JS SDK docs yet, I think starting with plain javascript and react is a good step.

JonasBa commented:
Ah, and I forgot to link to sourcemap setup - we need these to be configured

JonasBa commented:
This should not be merged until the next JS SDK release (I will update the min required version then)

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+#### Install
+
+Our browser profiling integration is built upon the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it is therefor in beta stage and is likely to only to move out as the official spec progresses. There are obviously risks with using a beta package - we've made sure that the SDK gracefully degrades in unsupported environments.
```
```suggestion
Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses. There are some risks involved with using a beta package. We've made sure that the SDK gracefully degrades in unsupported environments.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+Our browser profiling integration is built upon the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it is therefor in beta stage and is likely to only to move out as the official spec progresses. There are obviously risks with using a beta package - we've made sure that the SDK gracefully degrades in unsupported environments.
+
+Please note that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherintly be biased towards that demographic - this is something you need to be aware of when making decisions.
```
```suggestion
Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/2.configure-document-policy.md:
```diff
@@ -0,0 +1,19 @@
+
+#### Add Document-Policy:js-profiling header
+
+In order to enable the profiler to start, the document response header has to include a header key `Document-Policy` with value of `js-profiling`.
```
```suggestion
In order for the profiler to start, the document response header has to include a `Document-Policy` header key with the `js-profiling` value.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/2.configure-document-policy.md:
```diff
@@ -0,0 +1,19 @@
+
+In order to enable the profiler to start, the document response header has to include a header key `Document-Policy` with value of `js-profiling`.
+
+Depending on your server, how you do this will vary. If you are using a server like express, you can do this like via the [response.set](https://expressjs.com/en/4x/api.html#res.set) function.
```
```suggestion
How you do this will depend on your server. If you're using a server like express, you'll be able to use the [response.set](https://expressjs.com/en/4x/api.html#res.set) function.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/3.configure.md:
```diff
@@ -0,0 +1,40 @@
+
+Configuration should happen as early as possible in your application's lifecycle.
+
+Once this is done, Sentry's JavaScript SDK captures all unhandled exceptions and transactions.
```
```suggestion
Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+#### Install
+
+Our browser profiling integration is built upon the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it is therefor in beta stage and is likely to only to move out as the official spec progresses. There are obviously risks with using a beta package - we've made sure that the SDK gracefully degrades in unsupported environments.
```
```suggestion
Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses. There are some risks involved with using a beta package. We've made sure that the SDK gracefully degrades in unsupported environments.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+Our browser profiling integration is built upon the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it is therefor in beta stage and is likely to only to move out as the official spec progresses. There are obviously risks with using a beta package - we've made sure that the SDK gracefully degrades in unsupported environments.
+
+Please note that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherintly be biased towards that demographic - this is something you need to be aware of when making decisions.
```
```suggestion
Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/2.configure-document-policy.md:
```diff
@@ -0,0 +1,19 @@
+
+#### Add Document-Policy:js-profiling header
+
+In order to enable the profiler to start, the document response header has to include a header key `Document-Policy` with value of `js-profiling`.
```
```suggestion
In order for the profiler to start, the document response header has to include a `Document-Policy` header key with the `js-profiling` value.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/2.configure-document-policy.md:
```diff
@@ -0,0 +1,19 @@
+
+In order to enable the profiler to start, the document response header has to include a header key `Document-Policy` with value of `js-profiling`.
+
+Depending on your server, how you do this will vary. If you are using a server like express, you can do this like via the [response.set](https://expressjs.com/en/4x/api.html#res.set) function.
```
```suggestion
How you do this will depend on your server. If you're using a server like express, you'll be able to use the [response.set](https://expressjs.com/en/4x/api.html#res.set) function.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/3.configure.md:
```diff
@@ -0,0 +1,40 @@
+
+Configuration should happen as early as possible in your application's lifecycle.
+
+Once this is done, Sentry's JavaScript SDK captures all unhandled exceptions and transactions.
```
```suggestion
Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
```

lizokm reviewed (approved):
Made some language suggestions :)

JonasBa commented:
You are the best, thank you @lizokm!

JonasBa commented:
@lizokm we may have some more edits as we probably want to link to sourcemaps docs from some of these pages. I'm going to wait for @indragiek to give it a read first to get his thoughts and will let you know if we need another review!

indragiek commented on src/wizard/javascript/profiling-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses. There are some risks involved with using a beta package. We've made sure that the SDK gracefully degrades in unsupported environments.
+
+Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic.
```
We could also link to this status page which shows whether other browsers are adopting it: https://chromestatus.com/feature/5170190448852992

indragiek reviewed (approved):
Looks great! One comment.

JonasBa commented:
@lizokm @indragiek I re-read the version and reworded the Install step instructions as they read poorly and were repetitive. 

I also added the profiling doc to global instructions for browser javascript

shanamatthews commented:
@JonasBa - Liza's approval is plenty, but let me know if you'd like my eyes on this too :)

JonasBa commented:
@shanamatthews yeah I figured, we just did some changes since the last time she approved it so I wanted to get one more look. If you have time to review it I'd really appreciate it :). You can just review my last commit, it's the one that includes changes since Liza reviewed it

lizokm commented on src/platforms/javascript/common/profiling/index.mdx:
```diff
@@ -0,0 +1,76 @@
+
+Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses and gains adoption. As with any beta package, there are risks involved in using it - see platform [status](https://chromestatus.com/feature/5170190448852992)
+
+Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic - this is something you need consider if you are basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and wou ...
```
```suggestion
Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic. This is something you'll need to consider if you're basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and would like to see it gain further adoption, please consider supporting the spec at the official [WICG repository](https://github.com/WICG/js-self-profiling).
```

lizokm commented on src/platforms/javascript/common/profiling/index.mdx:
```diff
@@ -0,0 +1,76 @@
+
+Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic - this is something you need consider if you are basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and wou ...
+
+In order to get started with browser profiling, you will need to:
+
+- Install the @sentry/browser SDK
+- Configure document response header to include `Document-Policy: js-profiling`
+- Configure the SDK to use the BrowserProfilingIntegration and set profilesSampleRate
```
```suggestion
In order to get started with browser profiling, you'll need to:

- Install the @sentry/browser SDK
- Configure the document response header to include `Document-Policy: js-profiling`
- Configure the SDK to use the BrowserProfilingIntegration and set profilesSampleRate
```

lizokm commented on src/platforms/javascript/common/profiling/index.mdx:
````diff
@@ -0,0 +1,76 @@
+});
+```
+
+#### Configure
+
+Configuration should happen as early as possible in your application's lifecycle.
+
+Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
````
```suggestion
## Configure

Configuration should happen as early as possible in your application's lifecycle. Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses and gains adoption. As with any beta package, there are risks involved in using it - see platform [status](https://chromestatus.com/feature/5170190448852992)
+
+Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic - this is something you need consider if you are basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and wou ...
```
```suggestion
Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic. This is something you'll need to consider if you're basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and would like to see it gain further adoption, please consider supporting the spec at the official [WICG repository](https://github.com/WICG/js-self-profiling).
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic - this is something you need consider if you are basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and wou ...
+
+Install our JavaScript browser SDK using either `yarn` or `npm`, the minimum version that supports profiling is **7.60.0**
```
```suggestion
Install our JavaScript browser SDK using either `yarn` or `npm`, the minimum version that supports profiling is **7.60.0**.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/1.install.md:
```diff
@@ -0,0 +1,22 @@
+type: framework
+---
+
+#### Install
```
```suggestion
## Install
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/2.configure-document-policy.md:
```diff
@@ -0,0 +1,19 @@
+type: framework
+---
+
+#### Add Document-Policy:js-profiling header
```
```suggestion
## Add Document-Policy: js-profiling header
```

lizokm commented on src/wizard/javascript/profiling-onboarding/javascript/3.configure.md:
```diff
@@ -0,0 +1,40 @@
+type: framework
+---
+
+#### Configure
+
+Configuration should happen as early as possible in your application's lifecycle.
+
+Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
```
```suggestion
## Configure

Configuration should happen as early as possible in your application's lifecycle. Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/1.install.md:
```diff
@@ -0,0 +1,22 @@
+type: framework
+---
+
+#### Install
+
+Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses and gains adoption. As with any beta package, there are risks involved in using it - see platform [status](https://chromestatus.com/feature/5170190448852992)
```
```suggestion
## Install

Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses and gains adoption. As with any beta package, there are risks involved in using it - see platform [status](https://chromestatus.com/feature/5170190448852992).
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+Because our browser profiling integration is built on top of the profiler exposed by the [JS Self-Profiling API](https://wicg.github.io/js-self-profiling/), it's in beta and will likely only move out once the official spec progresses and gains adoption. As with any beta package, there are risks involved in using it - see platform [status](https://chromestatus.com/feature/5170190448852992)
+
+Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic - this is something you need consider if you are basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and wou ...
```
```suggestion
Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic. This is something you'll need to consider if you're basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and would like to see it gain further adoption, please consider supporting the spec at the official [WICG repository](https://github.com/WICG/js-self-profiling).
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/1.install.md:
```diff
@@ -0,0 +1,22 @@
+
+Please note, that since profiling API is currently only implemented in Chromium based browsers, the profiles collected will inherently be biased towards that demographic - this is something you need consider if you are basing your decisions on the data collected. We hope that as the API gains adoption, other browsers will implement it as well. If you find browser profiling feature helpful and wou ...
+
+Install our JavaScript browser SDK using either `yarn` or `npm`, the minimum version that supports profiling is **7.60.0**
```
```suggestion
Install our JavaScript browser SDK using either `yarn` or `npm`, the minimum version that supports profiling is **7.60.0**.
```

lizokm commented on src/wizard/javascript/profiling-onboarding/react/3.configure.md:
```diff
@@ -0,0 +1,40 @@
+type: framework
+---
+
+#### Configure
+
+Configuration should happen as early as possible in your application's lifecycle.
+
+Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
```
```suggestion
## Configure

Configuration should happen as early as possible in your application's lifecycle. Once this is done, Sentry's JavaScript SDK will capture all unhandled exceptions and transactions.
```

lizokm reviewed (approved):
Just a few more small tweaks.
